runner/src/scenario/ceramic/event_id_sync.rs

```diff
@@ -0,0 +1,182 @@
+    let cli = CeramicHttpClient::new(creds.signer);
+    let redis_cli = get_redis_client().await?;
+
+    let test_start = Transaction::new(Arc::new(move |user| {
+        Box::pin(setup(
+            user,
+            cli.clone(),
+            redis_cli.clone(),
+            ipfs_addr.clone(),
+        ))
+    }))
+    .set_name("setup")
+    .set_on_start();
+
+    let create_new_event = transaction!(create_new_event).set_name(CREATE_EVENT_TX_NAME);
+
+    Ok(scenario!("EventIDSync")
+        .register_transaction(test_start)
+        .register_transaction(create_new_event))
+}
+
+/// One user on one node creates a model.
+/// One user on each node subscribes to the model via Recon
+#[instrument(skip_all, fields(user.index = user.weighted_users_index), ret)]
+async fn setup(
+    user: &mut GooseUser,
+    cli: CeramicClient,
+    redis_cli: redis::Client,
+    ipfs_peer_addr: Url,
+) -> TransactionResult {
+    let mut conn = redis_cli.get_async_connection().await.unwrap();
+    let model_id = if should_request_events() && is_first_user() {
+        info!("creating model for event ID sync test");
+        let small_model = match ModelDefinition::new::<models::SmallModel>(
+            "load_test_small_model",
+            ModelAccountRelation::List,
+        ) {
+            Ok(model) => model,
+            Err(e) => {
+                tracing::error!("failed to create model: {}", e);
+                panic!("failed to create model: {}", e);
+            }
+        };
+        let model_id = match setup_model(user, &cli, small_model).await {
+            Ok(model_id) => model_id,
+            Err(e) => {
+                tracing::error!("failed to setup model: {:?}", e);
+                return Err(e);
+            }
+        };
+        set_model_id(&mut conn, &model_id, MODEL_ID_KEY).await;
+        model_id
+    } else {
+        get_model_id(&mut conn, MODEL_ID_KEY).await
+    };
+
+    tracing::debug!(%model_id, "syncing model");
+
+    let path = format!("/ceramic/subscribe/model/{}?limit=1", model_id);
+    let user_data = ModelReuseLoadTestUserData {
+        cli,
+        redis_cli,
+        model_id,
+    };
+    user.set_session_data(user_data);
+    user.base_url = Some(ipfs_peer_addr); // Recon is only available on IPFS address right now
+
+    let request_builder = user
+        .get_request_builder(&GooseMethod::Get, &path)?
+        .timeout(Duration::from_secs(5));
+    let req = GooseRequest::builder()
+        .set_request_builder(request_builder)
+        .expect_status_code(200)
+        .build();
+
+    let _goose = user.request(req).await?;
+
+    Ok(())
+}
+
+/// Generate a random event that the nodes are interested in. Only one node should create but all
+/// users do it so that we can generate a lot of events.
+async fn create_new_event(user: &mut GooseUser) -> TransactionResult {
+    if !should_request_events() {
+        tokio::time::sleep(Duration::from_millis(500)).await;
```

Comment:
Am I correct in guessing this is so we do not spin the CPU doing nothing for the node that is not writing data? If so could you add a comment to that effect?

Reply:
Yep, added a comment.